utils/src/string_operations.cpp

```diff
@@ -95,15 +97,13 @@ std::string_view next_nonblank_sequence(std::string_view s)
 
 std::string& to_upper(std::string& s)
 {
-    for (auto& c : s)
-        c = static_cast<char>(std::toupper((unsigned char)c));
+    std::transform(s.begin(), s.end(), s.begin(), [](unsigned char c) { return upper_cased[c]; });
     return s;
 }
 
 std::string to_upper_copy(std::string s)
 {
-    for (auto& c : s)
-        c = static_cast<char>(std::toupper((unsigned char)c));
+    to_upper(s);
     return s;
```

Comment:
How about this?
```suggestion
    return to_upper(std::move(s));
```

Reply:
That would not build - rvalue ref does not bind to (non-cont) lvalue ref, and if it did, it would make a copy on return because to_upper returns lvalue ref.